src/cborparser.c

```diff
@@ -1093,7 +1093,8 @@ static uintptr_t iterate_memcmp(char *s1, const uint8_t *s2, size_t len)
 
 static uintptr_t iterate_memcpy(char *dest, const uint8_t *src, size_t len)
 {
-    return (uintptr_t)memcpy(dest, src, len);
+    memcpy(dest, src, len);
+    return (uintptr_t)dest;
```

Comment:
I don't want this change. It's supposed to be the same thing, but this causes the `iterate_memcpy` function to require a stack frame, save the `dest` pointer somewhere, perform the call and then restore the value. The current implementation is a simple tail-call jump to `memcpy`.

What does you compiler mean with "can't be guaranteed without loss of information"? The definition of `uintptr_t`  says (C11 7.20.1.4):
> The following type designates an unsigned integer type with the property that any valid pointer to **void** can be converted to this type, then converted back to pointer to **void**, and the result will compare equal to the original pointer:

What compiler and architecture was this emitted for?

Reply:
arm eabi (gcc)
you're indeed right... that's a tricky one